analyzers/src/SonarAnalyzer.Common/Rules/SelfAssignmentBase.cs

```diff
@@ -26,16 +26,17 @@ namespace SonarAnalyzer.Rules
 {
     public abstract class SelfAssignmentBase : SonarDiagnosticAnalyzer
```

Comment:
<TSyntaxKind> where TSyntaxKind : struct

Reply:
The generic version is not needed by this rule

Reply:
I don't understand what you mean. Can you elaborate more on the original suggestion?

Reply:
Ah, I got it. It's about using `SonarDiagnosticAnalyzer<TSyntaxKind>`, I was looking at `SelfAssignmentBase<TSyntaxKind>` only.
Ok, nice.